include/unifex/create.hpp

```diff
@@ -1,197 +1,303 @@
- *   to \c fn , which will complete the receiver passed to \c connect in turn.
- */
-template <typename... ValueTypes>
-inline constexpr _create::_cpo::_fn<ValueTypes...> create {};
-
-using _create::void_cast;
-} // namespace unifex
-
-#include <unifex/detail/epilogue.hpp>
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/manual_lifetime.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _create {
+
+template <typename Receiver, typename Fn, typename... Context>
+struct _op {
+  struct type;
+  struct _receiver {
+    explicit _receiver(type* op) noexcept(
+        std::is_nothrow_move_constructible_v<Receiver>)
+      : op_(op) {}
+
+    template(typename... Ts)                     //
+        (requires receiver_of<Receiver, Ts...>)  //
+        void set_value(Ts&&... ts) noexcept(
+            is_nothrow_receiver_of_v<Receiver, Ts...>) {
+      unifex::set_value(std::move(get_receiver()), (Ts &&) ts...);
+    }
+
+    template(typename Error)                  //
+        (requires receiver<Receiver, Error>)  //
+        void set_error(Error&& error) noexcept {
+      unifex::set_error(std::move(get_receiver()), (Error &&) error);
+    }
+
+    void set_done() noexcept {
+      unifex::set_done(std::move(get_receiver()));
+    }
+
+  private:
+    Receiver& get_receiver() const { return op_->get_receiver(); }
+    // Forward other receiver queries
+    template(typename CPO)                          //
+        (requires is_receiver_query_cpo_v<CPO> AND  //
+             is_callable_v<CPO, const Receiver&>)   //
+        friend auto tag_invoke(CPO cpo, const _receiver& self) noexcept(
+            is_nothrow_callable_v<CPO, const Receiver&>)
+            -> callable_result_t<CPO, const Receiver&> {
+      return std::move(cpo)(self.get_receiver());
+    }
+    type* op_;
+  };
+  struct type {
+    using context_t = std::tuple<Context...>;
+    using state_t = callable_result_t<Fn, _receiver&, Context&...>;
+
+    ~type() {
+      if (started_) {
+        state_.destruct();
+      }
+    }
+    explicit type(Receiver rec, Fn fn, context_t ctx) noexcept(
+        std::is_nothrow_move_constructible_v<Fn>&&
+            std::is_nothrow_move_constructible_v<context_t>)

```

Comment:
Missing `is_notrhow_move_constructible_v<Receiver>`?
```suggestion
        std::is_nothrow_move_constructible_v<Receiver>&&
        std::is_nothrow_move_constructible_v<Fn>&&
            std::is_nothrow_move_constructible_v<context_t>)

```